src/models/MyAccount.ts

```diff
@@ -94,32 +94,42 @@ export default class MyAccount {
     return bibData
   }
 
+  /**
+   * getBibVarFields
+   * Reads varFields of a item-level hold's bib data to return if it is a research item,
+   * and if it is owned by NYPL.
+   */
+
+  static getBibVarFields(bibFields) {
```

Comment:
this snippet can be included here:
```
// Bib level hold won't have varFields, and we know it will be circ and NYPL owned.
      if (!bibFields.varFields) {
        isResearch = false
        isNyplOwned = true
      }
```
The pattern I'm going for here is pretty much like a switch - eg returning within each if statement. A little more verbose, but tends to be easier to read down the line. So the snippet above can be the first part of this method, and you can return the values straight away.

Reply:
This seems to be just recreating the nested if statement? I'm unclear what we're trying to avoid at this point

Reply:
this is what I'm getting at:
```
if no varfields
	  return isnypl: true isResearch: false

else if no 910 field
	  return isResearch: true isNypl: false (based on !isResearch from line 121)
	
else 
	  calulate isResearch, isNypl
	  return isResearch isNypl
```

otherwise you have these values being calculated in different places. also, as it is now, we have the info we need to return as soon as we know something doesn't have a 910 field. there's no need to couple that logic with the logic for records with 910 fields.

Reply:
Ahhhhh okay. i think i got it thank you